discovery/uyuni/uyuni.go

```diff
@@ -0,0 +1,319 @@
+	var systemGroupsInfos []struct {
+		SystemID     int             `xmlrpc:"id"`
+		SystemGroups []systemGroupID `xmlrpc:"system_groups"`
+	}
+	err := rpcclient.Call("system.listSystemGroupsForSystemsWithEntitlement", []interface{}{token, entitlement}, &systemGroupsInfos)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[int][]systemGroupID)
+	for _, systemGroupsInfo := range systemGroupsInfos {
+		result[systemGroupsInfo.SystemID] = systemGroupsInfo.SystemGroups
+	}
+	return result, nil
+}
+
+// GetSystemNetworkInfo lists client FQDNs.
+func getNetworkInformationForSystems(rpcclient *xmlrpc.Client, token string, systemIDs []int) (map[int]networkInfo, error) {
+	var networkInfos []networkInfo
+	err := rpcclient.Call("system.getNetworkForSystems", []interface{}{token, systemIDs}, &networkInfos)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[int]networkInfo)
+	for _, networkInfo := range networkInfos {
+		result[networkInfo.SystemID] = networkInfo
+	}
+	return result, nil
+}
+
+// Get endpoints information for given systems
+func getEndpointInfoForSystems(
+	rpcclient *xmlrpc.Client,
+	token string,
+	systemIDs []int,
+) ([]endpointInfo, error) {
+	var endpointInfos []endpointInfo
+	err := rpcclient.Call(
+		"system.monitoring.listEndpoints",
+		[]interface{}{token, systemIDs}, &endpointInfos)
+	if err != nil {
+		return nil, err
+	}
+	return endpointInfos, err
+}
+
+// NewDiscovery returns a uyuni discovery for the given configuration.
+func NewDiscovery(conf *SDConfig, logger log.Logger) *Discovery {
+	d := &Discovery{
+		interval: time.Duration(conf.RefreshInterval),
+		sdConfig: conf,
+		logger:   logger,
+	}
+	d.Discovery = refresh.NewDiscovery(
+		logger,
+		"uyuni",
+		time.Duration(conf.RefreshInterval),
+		d.refresh,
+	)
+	return d
+}
+
+func (d *Discovery) getEndpointLabels(
+	endpoint endpointInfo,
+	systemGroupIDs []systemGroupID,
+	networkInfo networkInfo,
+) model.LabelSet {
+
+	var addr string
+	managedGroupNames := getSystemGroupNames(systemGroupIDs)
+	addr = fmt.Sprintf("%s:%d", networkInfo.Hostname, endpoint.Port)
+
+	result := model.LabelSet{
+		model.AddressLabel:       model.LabelValue(addr),
+		uyuniLabelMinionHostname: model.LabelValue(networkInfo.Hostname),
+		uyuniLabelPrimaryFQDN:    model.LabelValue(networkInfo.PrimaryFQDN),
+		uyuniLablelSystemID:      model.LabelValue(fmt.Sprintf("%d", endpoint.SystemID)),
+		uyuniLablelGroups:        model.LabelValue(strings.Join(managedGroupNames, d.sdConfig.Separator)),
+		uyuniLablelEndpointName:  model.LabelValue(endpoint.EndpointName),
+		uyuniLablelExporter:      model.LabelValue(endpoint.ExporterName),
+		uyuniLabelProxyModule:    model.LabelValue(endpoint.Module),
+		uyuniLabelMetricsPath:    model.LabelValue(endpoint.Path),
+	}
+
+	level.Debug(d.logger).Log("msg", "Configured target", "Labels", fmt.Sprintf("%+v", result))
```

Comment:
This should not be logged, there is an api endpoint for this.

Reply:
Removed.